enhancements/network/ovn-kubernetes-evpn.md

```diff
@@ -0,0 +1,429 @@
+```
+
+During development or tech preview, replace the FRR version with an upstream one
+one that can support the EVPN feature:
+
+```shell
+oc patch Network.operator.openshift.io cluster --type=merge -p='{"spec":{"managementState": "Unmanaged"}}'
+...
+oc set image -n openshift-frr-k8s ds/frr-k8s frr=quay.io/frrouting/frr:10.4.1 reloader=quay.io/frrouting/frr:10.4.1
+```
+
+Then, a cluster administrator proceeds with the OVN-Kubernetes [workflow][8] to
+configure EVPN. What follows is an example of that workflow.
+
+[8]: https://github.com/ovn-kubernetes/ovn-kubernetes/blob/4568650412f85cdcf7671c74d2e9b3bb67d3b366/docs/okeps/okep-5088-evpn.md#workflow-description
+
+1. Configure FRR-K8s defining the routers to establish the underlay BGP peering
+   and the routers that correspond to the desired IP-VRFs:
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  name: evpn
+  namespace: openshift-frr-k8s
+  labels:
+    advertise: evpn
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      neighbors:
+      - address: 172.18.0.5
+        asn: 64513
+```
+
+2. Configure a VTEP
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: VTEP
+metadata:
+  name: evpn-vtep
+spec:
+  cidr: 100.64.0.0/24
+  mode: managed
+```
+
+3. Configure RouteAdvertisements targeting the reference FRRConfiguration to use
+   and the CUDNs to be advertised in an EVPN configuration:
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: RouteAdvertisements
+metadata:
+  name: advertise-cudns-evpn
+spec:
+  targetVRF: auto
+  advertisements:
+  - PodNetwork
+  networkSelectors:
+  - networkSelectionType: ClusterUserDefinedNetworks
+    clusterUserDefinedNetworkSelector:
+      networkSelector:
+        matchLabels:
+          advertise: evpn
+  frrConfigurationSelector:
+    matchLabels:
+      advertise: evpn
+  nodeSelector: {}
+```
+
+4. Configure the L2 and/or L3 CUDNs with EVPN
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: l3-cudn
+  labels:
+    advertise: evpn
+spec:
+  namespaceSelector:
+    matchLabels:
+      kubernetes.io/metadata.name: l3-cudn
+  topology: Layer3
+  layer3:
+    role: Primary
+    subnets:
+      - cidr: "22.100.0.0/16"
+        hostSubnet: 24
+  transport: EVPN
+  evpnConfiguration:
+    vtep: evpn-vtep
+    ipVRF:
+      vni: 100
+---
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: l2-cudn
+  labels:
+    advertise: evpn
+spec:
+  namespaceSelector:
+    matchLabels:
+      kubernetes.io/metadata.name: l2-cudn
+  network:
+    topology: Layer2
+    layer2:
+      role: Primary
+      subnets:
+      - "22.100.0.0/16"
+  transport: EVPN
+  evpnConfiguration:
+    vtep: evpn-vtep
+    macVRF:
+      vni: 111
+    ipVRF:
+      vni: 101
+```
+
+### API Extensions
+
+There are no changes required to OCP specific APIs.
+
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+No special considerations for hosted clusters.
```

Comment:
but do we plan to support this feature on HCP given we don't claim BGP support on HCP?
my understanding is we are only going to support this feature (at least initially) on OCP - baremetal (not even vsphere) right?

Reply:
Added note